Api/ResourceInterface.php

```diff
@@ -90,4 +90,11 @@ public function getValidationGroups();
      * @return string
      */
     public function getShortName();
+
+    /**
+     * Gets the custom pluralized name of the resource.
+     *
+     * @return string
+     */
+    public function getPluralizedName();
```

Comment:
I would add this method to the `RoutedResourceInterface` instead and test if the resource is an instance of this interface to prevent BC break (this new interface was not released yet so it's OK to add new method to it - and it makes sens here as the `pluralizedName` is used only for routing).

Btw, I would give another name for this property such as `basePath` or something similar because some people may want to name their URLs in singular.